summoner-cli/src/Summoner/Template/Nix.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE QuasiQuotes    #-}
+
+module Summoner.Template.Nix
+       ( nixFiles
+       ) where
+
+import NeatInterpolation (text)
+
+import Summoner.Default (defaultGHC)
+import Summoner.GhcVer (GhcVer (..))
+import Summoner.Settings (NixPkgSet (..), Settings (..), defaultNixPkgSet)
+import Summoner.Tree (TreeFs (..))
+
+import qualified Data.Text as T
+
+nixCompiler :: GhcVer -> Text
+nixCompiler = \case
+    Ghc7103 -> "ghc7103"
+    Ghc801  -> "ghc802"
+    Ghc802  -> "ghc802"
+    Ghc822  -> "ghc822"
+    Ghc843  -> "ghc844"
+    Ghc844  -> "ghc844"
+    Ghc863  -> "ghc863"
+
+defaultNixCompiler :: Text
+defaultNixCompiler = nixCompiler defaultGHC
+
+nixFiles :: Settings -> [TreeFs]
+nixFiles Settings{..} =
+    [ shellNix
+    , releaseNix
+    , defaultNix
+    , nixDir
+    ]
+    where
+        defaultNix = File "default.nix" defaultNixT
+        shellNix = File "shell.nix" shellNixT
+        releaseNix = File "release.nix" releaseNixT
+
+        nixDir = Dir "nix" [fetchNixpkgsNix, nixpkgsNix, overridesNix]
+        fetchNixpkgsNix = File "fetchNixpkgs.nix" fetchNixpkgsNixT
+        nixpkgsNix = File "nixpkgs.nix" nixpkgsNixT
+        overridesNix = File "overrides.nix" overridesNixT
+
+        defaultNixT :: Text
+        defaultNixT =
+            [text|
+            { compiler ? "${defaultNixCompiler}"
+            , nixpkgs ? (import ./nix/nixpkgs.nix { inherit compiler; })
+            }:
+
+            with rec {
+              drv = nixpkgs.haskellPackages.${settingsRepo};
+            };
+
+            drv
+            |]
+
+        shellNixT :: Text
+        shellNixT = "(import ./default.nix {}).env" <> endLine
+
+        releaseNixT :: Text
+        releaseNixT = T.concat
+            [ "{"
+            , endLine
+            , T.intercalate endLine
```

Comment:
probably it's better to move int separate function, which will build this matrix?